### PR TITLE
MAC-166-Outer API for Reservation

### PR DESCRIPTION
**Pull request by ratheeshri1984**

### CODE REVIEW

File: src/EmployerAccounts/SFA.DAS.EmployerAccounts.Api.UnitTests/Controllers/Reservations/WhenGettingAllReservations.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        [MoqAutoData]
+        public async Task Then_Gets_Reservations_From_Mediator(
+            string accountId,
+            GetReservationsQueryResult mediatorResult,
+            [Frozen] Mock<IMediator> mockMediator,
+            [Greedy] ReservationController controller)
+        {
+            mockMediator
+                .Setup(mediator => mediator.Send(
+                    It.IsAny<GetReservationsQuery>(),
```
Comment (cofaulco) [inline src/EmployerAccounts/SFA.DAS.EmployerAccounts.Api.UnitTests/Controllers/Reservations/WhenGettingAllReservations.cs:30]: can we check that accountId is set correctly on the mediator request here?

**Review by cofaulco — CHANGES_REQUESTED** [review]
1 comment

File: src/EmployerAccounts/SFA.DAS.Approvals.FakeApis/ReservationsApiBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+using WireMock.RequestBuilders;
+using WireMock.ResponseBuilders;
+using WireMock.Server;
+
+namespace SFA.DAS.EmployerAccounts.FakeApis
+{
+    public class ReservationsApiBuilder
+    {
+        private readonly WireMockServer _server;
+        private readonly Fixture _fixture;
```
Comment (cofaulco) [inline src/EmployerAccounts/SFA.DAS.Approvals.FakeApis/ReservationsApiBuilder.cs:15]: created this class level Fixture and then not used it, instead opting for `new Fixture()` in the `WithReservations()` method

**Review by cofaulco — CHANGES_REQUESTED** [review]
issue picked up by sonarcloud

File: src/EmployerAccounts/SFA.DAS.Approvals.FakeApis/ReservationsApiBuilder.cs (hunk 2 above)
Comment (ratheeshri1984) [reply src/EmployerAccounts/SFA.DAS.Approvals.FakeApis/ReservationsApiBuilder.cs:15]: done

File: src/EmployerAccounts/SFA.DAS.EmployerAccounts.Api.UnitTests/Controllers/Reservations/WhenGettingAllReservations.cs (hunk 1 above)
Comment (ratheeshri1984) [reply src/EmployerAccounts/SFA.DAS.EmployerAccounts.Api.UnitTests/Controllers/Reservations/WhenGettingAllReservations.cs]: done

**Review by cofaulco — DISMISSED** [review]

File: src/EmployerAccounts/SFA.DAS.EmployerAccounts.Api.UnitTests/Controllers/Reservations/WhenGettingAllReservations.cs (hunk 1 above)
Comment (ratheeshri1984) [reply src/EmployerAccounts/SFA.DAS.EmployerAccounts.Api.UnitTests/Controllers/Reservations/WhenGettingAllReservations.cs]: done

File: src/EmployerAccounts/SFA.DAS.Approvals.FakeApis/ReservationsApiBuilder.cs (hunk 2 above)
Comment (ratheeshri1984) [reply src/EmployerAccounts/SFA.DAS.Approvals.FakeApis/ReservationsApiBuilder.cs:17]: done

**Pull request merged**